src/Lumi/Components/Button.purs

```diff
@@ -22,17 +22,18 @@ import React.Basic.DOM (CSS, css, unsafeCreateDOMComponent)
 import React.Basic.DOM as R
 import React.Basic.Events (EventHandler)
 
+data ButtonState = Enabled | Disabled | Loading
+
 type CommonButtonProps rest =
   { accessibilityLabel :: Nullable String
   , color :: Nullable ColorName
-  , disabled :: Boolean
   , onPress :: EventHandler
   , size :: Size
   , style :: CSS
   , testId :: Nullable String
   , title :: String
   , type :: String
-  , loading :: Boolean
+  , buttonState :: Maybe ButtonState
```

Comment:
Did you consider making this a `ButtonState` rather than a `Maybe ButtonState`? I'm not sure what `Nothing` represents here.